File: 2015/python/day01.py
def solve(input):
  p1, p2 = 0, -1
  p = 0
  for i, x in enumerate(input):
    if x == '(': p += 1 
    else: p -= 1
    if p == -1 and p2 == -1:
      p2 = i + 1
  p1 = p
  print("Part 1:", p1)
  print("Part 2:", p2)

# solve(")")
# solve("()())")

import sys
locs = [x[x.find('=')+1:] for x in sys.argv if x.startswith("--input-location")] \
  + ['../inputs/day01input.txt']
solve(open(locs[0]).read())

File: 2015/rust/day01.rs
use std::fs;

fn solve(input: String) {
  let mut p1 = 0;
  let mut p2 = 0;
  for (i, x) in input.chars().enumerate() {
    if x == '(' { 
      p1 += 1;
    } else { 
      p1 -= 1;
    }
    if p1 == -1 && p2 == 0 {
      p2 = i + 1;
    }
  }
  
  println!("Part 1: {}", p1);
  println!("Part 2: {}", p2);
}

fn main() {
  let input = fs::read_to_string("../inputs/day01input.txt").unwrap();
  solve(input)
}